src/ast/identifier.rs

```diff
@@ -1,9 +1,9 @@
 use derive_more::Display;
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Display)]
-pub struct Identifier<'a>(pub &'a str);
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Display)]
+pub struct Identifier(pub String);
```

Comment:
I think given `Map` is probably better off using `String`, this can probably go back to how it is.

However, if we were to keep it, a way to achieve what you're wanting here without always needing to use heap allocated `String` is https://doc.rust-lang.org/std/borrow/enum.Cow.html It essentially can be either a borrowed strsing, or an owned string, only allocating an owned one if it needs to be.